From 14dca2f9fc4fd1a370cdf7d0d690472e1f56da8b Mon Sep 17 00:00:00 2001
From: ChangSeok Oh <shivamidow@gmail.com>
Date: Tue, 3 Sep 2013 13:51:29 +0200
Subject: [PATCH 2/9] [Texmap][GTK] The poster-circle doesn't appear.
 https://bugs.webkit.org/show_bug.cgi?id=106672

Patch by ChangSeok Oh <shivamidow@gmail.com> on 2013-01-21
Reviewed by Noam Rosenthal.

Source/WebCore:

We need to call setNeedsDisplay to redraw GraphicsLayer after we set drawsContent
a new value for the layer. Otherwise we can't get a chance for first drawing the layer.

Test: compositing/visibility/visibility-composited-animation.html

* platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::flushCompositingStateForThisLayerOnly):

LayoutTests:

A new pixel test is added. It tests if animated layer is drawn for the first scene.

* compositing/visibility/visibility-composited-animation-expected.png: Added.
* compositing/visibility/visibility-composited-animation-expected.txt: Added.
* compositing/visibility/visibility-composited-animation.html: Added.

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@140347 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 LayoutTests/ChangeLog                              | 13 +++++++
 .../visibility-composited-animation-expected.txt   |  2 +
 .../visibility-composited-animation.html           | 45 ++++++++++++++++++++++
 Source/WebCore/ChangeLog                           | 15 ++++++++
 .../graphics/texmap/TextureMapperLayer.cpp         |  3 ++
 5 files changed, 78 insertions(+)
 create mode 100644 LayoutTests/compositing/visibility/visibility-composited-animation-expected.txt
 create mode 100644 LayoutTests/compositing/visibility/visibility-composited-animation.html

diff --git a/Source/WebCore/ChangeLog b/Source/WebCore/ChangeLog
index a220029..16d902c 100644
--- a/Source/WebCore/ChangeLog
+++ b/Source/WebCore/ChangeLog
@@ -838,6 +838,21 @@
         * dom/Element.h:
         (Element):
 
+2013-01-21  ChangSeok Oh  <shivamidow@gmail.com>
+
+        [Texmap][GTK] The poster-circle doesn't appear.
+        https://bugs.webkit.org/show_bug.cgi?id=106672
+
+        Reviewed by Noam Rosenthal.
+
+        We need to call setNeedsDisplay to redraw GraphicsLayer after we set drawsContent
+        a new value for the layer. Otherwise we can't get a chance for first drawing the layer.
+
+        Test: compositing/visibility/visibility-composited-animation.html
+
+        * platform/graphics/texmap/TextureMapperLayer.cpp:
+        (WebCore::TextureMapperLayer::flushCompositingStateForThisLayerOnly):
+
 2013-01-15  Allan Sandfeld Jensen  <allan.jensen@digia.com>
 
         [Qt][CSS Shaders] Make custom filter render in Wk1 mode
diff --git a/Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp b/Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp
index d2a10aa..4024333 100644
--- a/Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp
+++ b/Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp
@@ -400,6 +400,9 @@ void TextureMapperLayer::flushCompositingStateSelf(GraphicsLayerTextureMapper* g
 
     m_size = graphicsLayer->size();
 
+    if ((changeMask & DrawsContentChange) && graphicsLayer->drawsContent())
+        graphicsLayer->setNeedsDisplay();
+
     if (changeMask & MaskLayerChange) {
        if (TextureMapperLayer* layer = toTextureMapperLayer(graphicsLayer->maskLayer()))
            layer->m_effectTarget = this;
-- 
1.8.3.1

